Frog masks [Feature] [ready]

Mercenaryblue opened the pull request:
[Changelogs]
:cl: 
add: Added frog masks. Reeeeeeeeee!!
/:cl:

<a href="https://imgur.com/ARIutFX"><img src="https://i.imgur.com/ARIutFX.png" title="source: imgur.com" /></a>
I'm sure he's a jolly fellow.
## What is it? 
Its a frog mask. You put on the mask, or your get the prod.
It modifies your speech and plays a sound every time you speak.

## Where is it? 
It is usually found in the costume machine with the other masks.

## Why?
Because it allows us to REEEEEEEEEEEEEEE!! non-stop. That's funny to me.

Mercenaryblue committed:
Frog Masks

added frog mask
added sprites

Mercenaryblue committed:
updating

recent changes

Mercenaryblue committed:
update m2k

now the only diff should be the frog mask.

Mercenaryblue committed:
update mk3

close but no cigar.

Cyberboss commented on code/modules/clothing/masks/miscellaneous.dm:
```diff
@@ -78,6 +78,43 @@
+	item_state = "frog"
+	flags_inv = HIDEFACE|HIDEHAIR|HIDEFACIALHAIR
+	w_class = WEIGHT_CLASS_SMALL
+	var/voicechange = 1
```
TRUE

Cyberboss commented on code/modules/clothing/masks/miscellaneous.dm:
```diff
@@ -78,6 +78,43 @@
+
+/obj/item/clothing/mask/frog/equipped(mob/user, slot) //when you put it on
+	var/mob/living/carbon/C = user
+	if(src == C.wear_mask)
```
reverse this check

Cyberboss commented on code/modules/clothing/masks/miscellaneous.dm:
```diff
@@ -78,6 +78,43 @@
+/obj/item/clothing/mask/frog/equipped(mob/user, slot) //when you put it on
+	var/mob/living/carbon/C = user
+	if(src == C.wear_mask)
+		playsound (usr, 'sound/effects/reee.ogg', 30, 1)
```
playsound(src,

Cyberboss commented on code/modules/clothing/masks/miscellaneous.dm:
```diff
@@ -78,6 +78,43 @@
+	return message
+
+obj/item/clothing/mask/frog/cursed
+	..()
```
wat

Mercenaryblue replied:
It fetch the info from /mask/frog/ then adds the cursed flag.

banthebantz replied:
wat

Cyberboss replied:
this isn't how types work

Qustinnus replied:
kek

Mercenaryblue replied:
okay. I'll just copy/paste it instead.

Qustinnus replied:
N O

Qustinnus replied:
just remove the ..() part lmao

Cyberboss commented on code/modules/clothing/masks/miscellaneous.dm:
```diff
@@ -78,6 +78,43 @@
+
+/obj/item/clothing/mask/frog/cursed/equipped(mob/user, slot)
+	var/mob/living/carbon/C = user
+	if(src == C.wear_mask)
```
reverse this check

JJRcop replied:
Yet works, it still does...
Change, however, it must...

Cyberboss commented on code/modules/clothing/masks/miscellaneous.dm:
```diff
@@ -78,6 +78,43 @@
+	var/mob/living/carbon/C = user
+	if(src == C.wear_mask)
+		to_chat(user, "<span class='warning'><B>[src] was cursed! Ree!!</B></span>")
+	..()
```
return ..()

QualityVan commented:
Can there be a version which isn't cursed? I kinda like the sprite.

Is there a cooldown for the sound it makes?

GunHog commented:
Is it possible for this to play the sound upon being hit, similar to how beating a clown makes it honk?

Mercenaryblue commented:
The regular version is not cursed. You can remove it at will.
I've tried to add a sound every time you attack, to no avail.
I'll try adding the on_hit sound.

Mercenaryblue committed:
Requested changes

Reeee!!

Mercenaryblue committed:
Requested changes Mk2

reshuffled the code a bit.
cursed info
return ..()

Mercenaryblue committed:
Revert "Requested changes Mk2"

This reverts commit 9f0b33e24f3315e3e9fa7225a3a06b6dfe56fc37.

Mercenaryblue committed:
Travis fix

Lets not re-shuffle, then.

Qustinnus commented on code/modules/clothing/masks/miscellaneous.dm:
```diff
@@ -78,6 +78,48 @@
+	var/mob/living/carbon/C = user
+	if(C.wear_mask == src)
+		playsound (src, 'sound/effects/reee.ogg', 30, 1)
+	..()
```
return ..()

Mercenaryblue committed:
return ..()

WJohn commented:
Would probably be better if the REEEEE noise was just an action button, because this sounds suuuuuuper annoying otherwise.

Qustinnus commented:
or atleast if it doesnt happen every time you talk

Iatots commented:
should definitely happen if you get hit, or at least batonned.

Mercenaryblue commented:
If you're worried about the sound spam, I could add a spam_flag

Mercenaryblue committed:
Voicebox and spam_flag

You can now turn off the voicebox on the normal frog mask. There's also a spam_flag to reduce the noise.

Mercenaryblue commented:
Added a voicebox option, and a sound spam flag.
It should be possible to talk normally with the mask on now.

Mercenaryblue committed:
delay

there's your magic number.

Anonmare commented:
Sorry no player references

Arianya commented:
headmins don't play, silly bunny

Mercenaryblue commented:
What player reference?

Iatots commented:
>some guy uses X in his forum avatar, now all future Xs introduced in the game are a player reference to him.

Dumb bunny.
Also, that's clearly a frog prince costume from runescape in the screenshot.

Bawhoppen commented:
better change my pic to a gun so we can't add any new guns

Mercenaryblue commented:
By that logic every clown feature I make is a player reference.

Qustinnus commented:
if i change my profile pic to a catgirl maybe i can get catgirls reverted because it would become a reference to me :^)

Armhulen commented:
ribbit

FantasticFwoosh commented:
Just took a second away from being deadmemed to share.

![](http://puu.sh/ydlMj/6cdd1bc931.jpg)

To think kor wanted me to pay him to make a frog suit. _2016, happier times..._

Cyberboss commented on code/modules/clothing/masks/miscellaneous.dm:
```diff
@@ -78,6 +78,64 @@
+	var/mob/living/carbon/C = user
+	if(C.wear_mask == src)
+		if(!spam_flag)
+			if(voicechange)
```
Merge these 3 ifs

Cyberboss commented on code/modules/clothing/masks/miscellaneous.dm:
```diff
@@ -78,6 +78,64 @@
+			if(voicechange)
+				playsound (src, 'sound/effects/reee.ogg', 30, 1)
+				spam_flag = TRUE
+				addtimer(CALLBACK(src, .proc/reset_spam), delay)
```
This bit of code should be separated into a proc

Cyberboss commented on code/modules/clothing/masks/miscellaneous.dm:
```diff
@@ -78,6 +78,64 @@
+			if(!spam_flag)
+				playsound (src, 'sound/effects/huuu.ogg', 30, 1)
+				spam_flag = TRUE
+				addtimer(CALLBACK(src, .proc/reset_spam), delay)
```
So as to not dupe it here

Cyberboss commented on code/modules/clothing/masks/miscellaneous.dm:
```diff
@@ -78,6 +78,64 @@
+			if(!spam_flag)
+				playsound (src, 'sound/effects/reee.ogg', 30, 1)
+				spam_flag = TRUE
+				addtimer(CALLBACK(src, .proc/reset_spam), delay)
```
Or here

Mercenaryblue committed:
optimization

also added a couple froggy lines.

Armhulen commented:
cmon this is ready!!!

ninjanomnom commented on code/modules/clothing/masks/miscellaneous.dm:
```diff
@@ -78,6 +78,66 @@
+/obj/item/clothing/mask/frog/proc/play_ree()
+	playsound (src, 'sound/effects/reee.ogg', 30, 1)
+	spam_flag = TRUE
+	addtimer(CALLBACK(src, .proc/reset_spam), delay)
```
rather than a timer have
`last_sound = world.time`
then in the play procs check
`if(world.time + delay > last_sound)`

ninjanomnom commented on code/modules/clothing/masks/miscellaneous.dm:
```diff
@@ -78,6 +78,66 @@
+/obj/item/clothing/mask/frog/proc/play_huu()
+	playsound (src, 'sound/effects/huuu.ogg', 30, 1)
+	spam_flag = TRUE
+	addtimer(CALLBACK(src, .proc/reset_spam), delay)
```
same as above

Mercenaryblue committed:
Frog Masks 1.0.6

Last changes - world timers.
removed a voice line.
doubled the delay

Shadowlight213 commented:
Oh my god this is the worst thing ever.
